ddtrace/contrib/pymongo/client.py

```diff
@@ -286,3 +309,19 @@ def _set_query_metadata(span, cmd):
         span.resource = "{} {} {}".format(cmd.name, cmd.coll, q)
     else:
         span.resource = "{} {}".format(cmd.name, cmd.coll)
+
+
+def set_query_rowcount(span, result=None, cursor=None):
+    search_key = "Batch"
+    try:
+        if not cursor:
+            cursor = result.docs[0].get("cursor")
+        # results returned in batches, get len of each batch
+        rowcount = sum([len(documents) for batch_key, documents in cursor.items() if search_key in batch_key])
+        span.set_metric(db.ROWCOUNT, rowcount)
+        return span
+
+    except Exception:
```

Comment:
Being more specific here about what type of exception is expected would make this code easier to maintain and learn.

Reply:
Changed this function and instead I just always pass the cursor object to the function. The try except was being used for my debugging purposes and I've also removed that.